guides/v2.2/javascript-dev-guide/javascript/requirejs.md

```diff
@@ -0,0 +1,103 @@
+---
+group: javascript-developer-guide
+subgroup: 1_Javascript
+title: RequireJS in Magento
+---
+
+This topic describes general concepts of how the [RequireJS library](http://requirejs.org){: target="_blank"} is used in Magento, with examples. Please refer to official RequireJS documentation for in-depth explanation.
+
+## RequireJS introduction {#requirejs-intro}
+
+RequireJS is a JavaScript file and module loader. It improves perceived page load times because it allows JavaScript to load in the background. In particular, it enables asynchronous JavaScript loading.
+
+## RequireJS in Magento {#requirejs-in-magento}
+
+### Configuration {#requirejs-config}
+
+All configuration is done in the `requirejs-config.js` file. It has a single root object `config` which contains the configuration options described below. All the configuration settings are optional and are used only when required. The following snippet is a sample `requirejs-config.js` describing the structure of the file.
+
+```javascript
+var config = {
+    'map': {...},
+    'paths': {...}
+    'shim': {...},
+    'config': {
+        'mixins': {...}
+    }
+}
+```
+
+#### map {#requirejs-config-map}
+
+The `map` configuration maps (connects) any real AMD modules that calls `define()`, to the specified alias. In the snippet below, `*` means all loaded RequireJS modules can use the specified alias.
+
+```javascript
+'map': {
+    '*': {
+        'alias': 'Vendor_Module/js/complex/path/amd-module'
+    }
+}
+```
+
+Now we can use our `Vendor_Module/js/complex/path/module` using `alias` in any RequireJS module or config file without needing to type the entire path.
+
```

Comment:
Could you please add a simple example of usage here?

Reply:
Added an example snippet, and also added an example `requirejs-config.js` present in Magento which has all configurations. Thanks!